lib/linter/linter.js

```diff
@@ -1850,16 +1865,27 @@ class Linter {
                                     }
                                 }
 
+                                // Apply meta.defaultOptions
+                                const slicedOptions = ruleOptions.slice(1);
+                                const mergedOptions = deepMergeArrays(rule.meta?.defaultOptions, slicedOptions);
+
+                                if (mergedOptions.length) {
+                                    ruleOptions = [ruleOptions[0], ...mergedOptions];
+                                }
```

Comment:
This could be wrapped in `else` of the above `if`, since in the `if` branch we're using options from the config which already have default options applies?